### ISSUE TITLE
JaQu - Db and DataSource 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#227]
```
Hi,

As discussed in the groups, I'm attaching a PooledDb implementation that will 
use a DataSource for database connections. 
I hope it will be of use.

Cheers

```

Original issue reported on code.google.com by `r...@mail.bg` on 26 Aug 2010 at 11:28

**Comment by GoogleCodeExporter** [comment]
```
Hi,

I don't really like "forceOpen" and using ThreadLocal. Also I would like to 
avoid adding a new class. If the application wants to use it, then that's OK, 
but as you already wrote in the javadoc: "In general open() is not slow - you 
should always call open()/close() on usage".

What about a new method in the Db class:

    public static Db open(DataSource ds) {
        try {
            return new Db(ds.getConnection());
        } catch (SQLException e) {
            throw convert(e);
        }
    }
```

Original comment by `thomas.t...@gmail.com` on 20 Sep 2010 at 6:32

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Sure, a new method is just fine:)

My idea was to create something like springs transaction per thread, but it is 
not so important.
```

Original comment by `r...@mail.bg` on 21 Sep 2010 at 7:08

**Comment by GoogleCodeExporter** [comment]
```
Implemented in version 1.2.144
```

Original comment by `thomas.t...@gmail.com` on 15 Oct 2010 at 6:37
- Changed state: **Fixed**
